packages/flutter_tools/test/integration.shard/test_driver.dart

```diff
@@ -183,8 +189,19 @@ abstract class FlutterTestDriver {
     );
   }
 
-  Future<int> quit() => _killGracefully();
+  /// Quits the currently running process.
+  @nonVirtual
+  Future<void> quit() async {
+    final int result = await _killGracefully();
+    if (result != 0) {
+      _debugPrint('Expected process to terminate gracefully, got exit code $result.');
+    }
+    if (!_hasExited) {
```

Comment:
My guess is that this could be flakey since it looks like there's potentially no delay between sending SIGKILL for the `_killForcefully` case, and doing this test.

Reply:
Ah I see `_killForcefully` returns `_process!.exitCode`, which in turn means it _must_ to have exited - but we could get unlucky on the microtask queue. Will add an event loop delay and a comment, thanks for the callout.